### ISSUE TITLE
when i exec "st2 packs install a" ,but b was installed!!!!why?

### ISSUE THREAD

**Issue by OMS-ZH** [StackStorm/st2#6012]
## SUMMARY

when i exec "st2 packs install a" ,but b was installed!

### STACKSTORM VERSION

Paste the output of ``st2 --version``:

##### OS, environment, install method

Post what OS you are running this on, along with any other relevant information/

- e.g. Docker, Vagrant, Kubernetes, etc. Describe how you installed ST2
- e.g. one-line install, custom install, etc -->

## Steps to reproduce the problem
**1.st2 pack install file:///opt/stackstorm/packs/sre_publish_handle**
      when i exec "st2 packs install a" ,but b was installed!
<img width="497" alt="企业微信截图_16911139138419" src="https://github.com/StackStorm/st2/assets/43334788/83d4fe4c-e85d-438c-a5fc-c3d010edcdce">

**2.sre_publish_handle/packs.yml**
   ref="sre_publish_handle"
<img width="680" alt="企业微信截图_16911134613219" src="https://github.com/StackStorm/st2/assets/43334788/9b1228a6-0b2c-434d-b26d-c50a35b81186">

**3.sre_alarm_handle/packs.yml**
   ref="sre_alarm_handle"
<img width="686" alt="企业微信截图_16911136107692" src="https://github.com/StackStorm/st2/assets/43334788/6a04b37b-0220-41c5-81f3-5af90f1f7fe8">
**4.st2 pack list**
  the sre_publish_handle was in pack list
<img width="668" alt="企业微信截图_16911139729942" src="https://github.com/StackStorm/st2/assets/43334788/0aceb6a2-900b-432b-b4f2-18fab8f00235">
**5. ls -al virtualenvs**
 i can`t find the documen that name is "sre_publish_handle" whit in virtualenvs
<img width="302" alt="企业微信截图_1691114017337" src="https://github.com/StackStorm/st2/assets/43334788/fe2c5263-3e3d-4df8-875d-a3cc96d45527">

## Expected Results

i think the pack of sre_pubish_handle should be installed finally. but i don`t know why......

**Comment by OMS-ZH** [comment]
the version is:
st2 3.8.0, on Python 3.8.10

**Comment by OMS-ZH** [comment]
> I think if your application package is written correctly and you put it in /opt/stackstorm/packs/, it will be installed when you execute st2 reload.

yes, when i execute st2 reload, the pack that in  /opt/stackstorm/packs will be install,but  the virtual environment doesn`t be created.so i need use "st2 packs install file://xxxx"  to  install  packs

**Comment by fdrab** [comment]
> yes, when i execute st2 reload, the pack that in /opt/stackstorm/packs will be install,but the virtual environment doesn`t be created.so i need use "st2 packs install file://xxxx" to install packs

you can also execute packs.setup_virtualenv. I do this in a orquesta workflow with newly created packs. I create template yaml files > packs.load > packs.setup_virtualenv > python script to add to RBAC > sudo st2-apply-rbac-definitions
